app/services/timeline_events/was_last_target_service.rb

```diff
@@ -0,0 +1,51 @@
+module TimelineEvents
+  class WasLastTargetService
+    def initialize(submission)
+      @submission = submission
+    end
+
+    def was_last_target?
+      return false unless startup.level == last_level
+
+      return false if final_milestone_targets.empty?
+
+      final_milestone_targets.all? do |target|
+        if target.team_target?
+          # Need to check for just one student.
+          target.status(student) == Targets::StatusService::STATUS_PASSED
+        else
+          # Need to check for each student in team.
+          students.all? do |s|
+            target.status(s) == Targets::StatusService::STATUS_PASSED
+          end
+        end
+      end
+    end
+
+    private
+
+    def student
+      @student ||= @submission.founders.first
+    end
+
+    def students
+      @students ||= @submission.founders
+    end
+
+    def startup
+      @startup ||= student.startup
+    end
+
+    def course
+      startup.course
+    end
+
+    def last_level
+      @last_level ||= course.levels.order(number: :desc).first
+    end
+
+    def final_milestone_targets
+      Target.joins(target_group: :level).where(target_groups: { milestone: true }, levels: { id: last_level.id })
```

Comment:
@harigopal: We should only check live targets. Tested with an archived target in the milestone group of final level and certificates were not issued on completion.

Reply:
Fixed in https://github.com/SVdotCO/pupilfirst/commit/16c45702d8a92823a22d28a85cb33fed48465211.